src/app/child-dev-project/notes/notes-related-to-entity/notes-related-to-entity.component.ts

```diff
@@ -103,6 +103,18 @@ export class NotesRelatedToEntityComponent
     };
   }
 
+  /**
+   * check if an entityId is already referenced in any array properties of the given note
+   */
+  private isAlreadyLinked(newNote: any, id: string): boolean {
+    for (const key in newNote) {
+      if (Array.isArray(newNote[key]) && newNote[key].includes(id)) {
```

Comment:
What if the other field, like "author", would only allow a single id, not an array?